server/game/Clocks/ChessClock.js

```diff
@@ -2,27 +2,68 @@ const Clock = require('./Clock');
 
 class ChessClock extends Clock {
     constructor(player, time) {
-        super(player, time);
+        super(player, time, 5);
         this.mode = 'stop';
         this.name = 'Chess Clock';
     }
 
+    pause() {
+        this.stop();
+    }
+
+    restart() {
+        this.start();
+    }
+
+    reset() {
+        this.stop();
+    }
+
     start() {
-        this.mode = 'down';
-        super.start();
+        if(!this.manuallyPaused) {
+            if(this.mode !== 'down') {
+                this.mode = 'down';
+                super.start();
+            }
+        }
     }
 
     stop() {
         super.stop();
         this.mode = 'stop';
     }
 
+    opponentStart() {
+
+    }
+
     timeRanOut() {
         this.player.game.addMessage('{0}\'s clock has run out', this.player);
         if(this.player.opponent && this.player.opponent.clock.timeLeft > 0) {
             this.player.game.recordWinner(this.player.opponent, 'clock');
         }
     }
+
+    updateTimeLeft(secs) {
+        if(this.timeLeft === 0 || secs < 0) {
+            return;
+        }
+        if(secs <= this.delayToStartClock) {
+            return;
+        }
+
+        secs = secs - this.delayToStartClock;
```

Comment:
This is what supports the delay.  We basically ignore the first 5 seconds of time taken during any window.